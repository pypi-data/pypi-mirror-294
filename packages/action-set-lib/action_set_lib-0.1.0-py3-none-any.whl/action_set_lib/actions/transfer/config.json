{
	"author":"ardio",
	"dependent_actions":["add_contact", "show_contact"],
	"system_permissions":["self_address", "redis", "ardio_contract"]
}