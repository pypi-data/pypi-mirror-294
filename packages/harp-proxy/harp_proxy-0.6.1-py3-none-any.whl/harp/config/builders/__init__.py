from .configuration import ConfigurationBuilder
from .system import System, SystemBuilder

__all__ = [
    "SystemBuilder",
    "ConfigurationBuilder",
    "System",
]
