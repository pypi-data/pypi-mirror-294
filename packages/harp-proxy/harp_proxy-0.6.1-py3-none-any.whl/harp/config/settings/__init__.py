from .base import BaseSetting, asdict, settings_dataclass
from .disabled import DisableableBaseSettings, DisabledSettings
from .from_file import FromFileSetting
from .lazy import Definition, Lazy

__all__ = [
    "BaseSetting",
    "Definition",
    "DisableableBaseSettings",
    "DisabledSettings",
    "FromFileSetting",
    "Lazy",
    "asdict",
    "settings_dataclass",
]
