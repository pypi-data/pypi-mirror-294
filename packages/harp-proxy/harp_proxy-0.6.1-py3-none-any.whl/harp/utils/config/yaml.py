from yaml import dump as yaml_dump
from yaml import load as yaml_load

try:
    from yaml import CDumper as Dumper
    from yaml import CLoader as Loader
except ImportError:
    from yaml import Dumper, Loader


def load(filename):
    with open(filename) as f:
        return yaml_load(f, Loader=Loader)


def dump(source):
    return yaml_dump(source, Dumper=Dumper)
