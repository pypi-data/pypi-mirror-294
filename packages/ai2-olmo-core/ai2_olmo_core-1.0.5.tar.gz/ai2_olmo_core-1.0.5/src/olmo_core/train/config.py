import os
import tempfile
from dataclasses import dataclass, field
from typing import Dict, Optional

import torch
import torch.distributed as dist
import torch.nn as nn
from torch.optim import Optimizer

from ..config import Config, DType
from ..data import DataCollator, MemMapDataset
from ..exceptions import OLMoConfigurationError
from ..io import is_url
from ..utils import get_default_device
from .callbacks import Callback
from .checkpoint import Checkpointer
from .trainer import LoadStrategy, Trainer
from .utils import Duration, DurationUnit


@dataclass
class TrainerConfig(Config):
    """
    A configuration class for easily building :class:`~olmo_core.train.trainer.Trainer` instances.
    """

    save_folder: str
    global_batch_size: int
    microbatch_size: int

    work_dir: Optional[str] = None
    load_path: Optional[str] = None
    load_strategy: LoadStrategy = LoadStrategy.if_available

    device: Optional[str] = None
    save_overwrite: bool = False
    max_duration: Duration = field(
        default_factory=lambda: Duration(value=1, unit=DurationUnit.epochs)
    )
    cancel_check_interval: int = 25
    hard_stop: Optional[Duration] = None
    metrics_collect_interval: int = 5
    callbacks: Dict[str, Callback] = field(default_factory=dict)
    fused_loss: bool = False
    z_loss_multiplier: Optional[float] = None
    autocast_precision: Optional[DType] = None
    data_seed: int = 0
    data_loader_workers: int = 0
    data_loader_prefetch_factor: Optional[int] = None

    def with_callback(self, name: str, callback: Callback) -> "TrainerConfig":
        """
        Add another callback.

        :param name: A name to assign the callback. Must be unique.
        :param callback: The callback to add.
        """
        if name in self.callbacks:
            raise OLMoConfigurationError(f"A callback with name '{name}' already exists")
        self.callbacks[name] = callback
        return self

    def build(
        self,
        model: nn.Module,
        optim: Optimizer,
        dataset: MemMapDataset,
        dp_process_group: Optional[dist.ProcessGroup] = None,
        checkpointer_pg: Optional[dist.ProcessGroup] = None,
    ) -> Trainer:
        """
        Build the corresponding trainer.

        :param model: The model to train.
        :param optim: The optimizer to use.
        :param dataset: The dataset to train on.
        """
        kwargs = self.as_dict(exclude_none=True, recurse=False)

        checkpointer = Checkpointer(
            save_overwrite=kwargs.pop("save_overwrite"),
            process_group=checkpointer_pg,
        )
        device = kwargs.pop("device", None)
        autocast_precision: Optional[DType] = kwargs.pop("autocast_precision", None)
        work_dir = kwargs.pop("work_dir", None)
        if work_dir is None:
            if not is_url(self.save_folder):
                work_dir = self.save_folder
            else:
                work_dir = os.path.join(tempfile.gettempdir(), os.path.basename(self.save_folder))

        collator = DataCollator(pad_token_id=dataset.pad_token_id)

        return Trainer(
            model=model,
            optim=optim,
            dataset=dataset,
            collator=collator,
            checkpointer=checkpointer,
            train_sequence_length=dataset.sequence_length,
            autocast_precision=None if autocast_precision is None else autocast_precision.as_pt(),
            work_dir=work_dir,
            device=torch.device(device) if device is not None else get_default_device(),
            dp_process_group=dp_process_group,
            **kwargs,
        )
