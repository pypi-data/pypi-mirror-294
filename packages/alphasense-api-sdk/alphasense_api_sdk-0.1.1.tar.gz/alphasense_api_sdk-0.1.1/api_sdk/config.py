from dataclasses import fields
from pathlib import Path
from typing import Dict, Optional
from warnings import simplefilter, warn

import toml
import json

from .client_generators.scalars import ScalarData
from .exceptions import ConfigFileNotFound, MissingConfiguration
from .settings import (
    ClientSettings,
    CommentsStrategy,
    GraphQLSchemaSettings,
    AuthSettings,
)
from .constants import (
    CONFIG_KEY,
    DEFAULT_CONFIG_FILE_PATH,
)

simplefilter("default", DeprecationWarning)


def get_config_file_path(file_name: str = DEFAULT_CONFIG_FILE_PATH) -> Path:
    """Get config file path. If not found raise exception."""
    directory = Path.cwd()
    while not directory.joinpath(file_name).exists():
        if directory == directory.parent:
            raise ConfigFileNotFound(f"Config file {file_name} not found.")
        directory = directory.parent
    return directory.joinpath(file_name).resolve()


def get_config_dict(config_file_name: Optional[str] = None) -> Dict:
    """Get config dict."""
    if config_file_name:
        config_file_path = get_config_file_path(config_file_name)
    else:
        config_file_path = get_config_file_path()

    return toml.load(config_file_path)


def get_client_settings(config_dict: Dict) -> ClientSettings:
    """Parse configuration dict and return ClientSettings instance."""
    section = get_config_section(config_dict).copy()
    settings_fields_names = {f.name for f in fields(ClientSettings)}
    try:
        section["scalars"] = {
            name: ScalarData(
                type_=data["type"],
                serialize=data.get("serialize"),
                parse=data.get("parse"),
                import_=data.get("import"),
            )
            for name, data in section.get("scalars", {}).items()
        }
    except KeyError as exc:
        raise MissingConfiguration(
            "Missing 'type' field for scalar definition"
        ) from exc

    try:
        if "include_comments" in section and isinstance(
            section["include_comments"], bool
        ):
            section["include_comments"] = (
                CommentsStrategy.TIMESTAMP.value
                if section["include_comments"]
                else CommentsStrategy.NONE.value
            )
            options = ", ".join(strategy.value for strategy in CommentsStrategy)
            warn(
                "Support for boolean 'include_comments' value has been deprecated "
                "and will be dropped in future release. "
                f"Instead use one of following options: {options}",
                DeprecationWarning,
                stacklevel=2,
            )

        return ClientSettings(
            **{
                key: value
                for key, value in section.items()
                if key in settings_fields_names
            }
        )
    except TypeError as exc:
        missing_fields = settings_fields_names.difference(section)
        raise MissingConfiguration(
            f"Missing configuration fields: {', '.join(missing_fields)}"
        ) from exc


def get_config_section(
    config_dict: Dict,
    config_key: str = "tool",
    section_name: str = "alphasense-api-codegen",
) -> Dict:
    """Get section from config dict."""
    cfg_key = config_key or CONFIG_KEY
    if cfg_key in config_dict and section_name in config_dict.get(cfg_key, {}):
        return config_dict[cfg_key][section_name]

    if section_name in config_dict:
        warn(
            f"Support for [{section_name}] section has been deprecated "
            "and will be dropped in future release. "
            f"Instead use [{cfg_key}.{section_name}].",
            DeprecationWarning,
            stacklevel=2,
        )
        return config_dict[section_name]

    raise MissingConfiguration(f"Config has no [{cfg_key}.{section_name}] section.")


def get_graphql_schema_settings(config_dict: Dict) -> GraphQLSchemaSettings:
    """Parse configuration dict and return GraphQLSchemaSettings instance."""
    section = get_config_section(config_dict)
    settings_fields_names = {f.name for f in fields(GraphQLSchemaSettings)}
    try:
        return GraphQLSchemaSettings(
            **{
                key: value
                for key, value in section.items()
                if key in settings_fields_names
            }
        )
    except TypeError as exc:
        missing_fields = settings_fields_names.difference(section)
        raise MissingConfiguration(
            f"Missing configuration fields: {', '.join(missing_fields)}"
        ) from exc


def get_auth_settings(config_dict: Dict) -> AuthSettings:
    """Parse configuration dict and return AsAuthSettings instance."""
    auth_cfg = get_config_section(
        config_dict, config_key="alphasense", section_name="auth"
    ).copy()
    cfg = get_config_section(config_dict)

    auth_cfg["schema_path"] = cfg.get("schema_path")
    if auth_cfg.get("schema_path") is None:
        auth_cfg["remote_schema_url"] = cfg.get("remote_schema_url")
    settings_fields_names = {f.name for f in fields(AuthSettings)}
    try:
        return AuthSettings(
            **{
                key: value
                for key, value in auth_cfg.items()
                if key in settings_fields_names
            }
        )
    except TypeError as exc:
        missing_fields = settings_fields_names.difference(auth_cfg)
        raise MissingConfiguration(
            f"Missing configuration fields: {', '.join(missing_fields)}"
        ) from exc
