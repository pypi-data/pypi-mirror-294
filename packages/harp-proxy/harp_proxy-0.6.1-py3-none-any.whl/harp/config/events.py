from typing import Awaitable, Callable

from rodi import Container, Services
from whistle import Event

from harp.asgi import ASGIKernel
from harp.controllers import ProxyControllerResolver
from harp.utils.network import Bind


class OnBindEvent(Event):
    """
    «Bind» event happens before the service container is resolved.

    It is the right place to define services that may not be resolvable yet (because their dependencies may, or may not,
    be defined already).

    """

    name = "harp.config.bind"

    def __init__(self, container: Container, settings):
        self.container = container
        self.settings = settings


EVENT_BIND = OnBindEvent.name

OnBindHandler = Callable[[OnBindEvent], Awaitable[None]]


class OnBoundEvent(Event):
    """
    «Bound» event happens after the service container is resolved, before the kernel creation.

    It is the right place to setup things that needs the container to be resolved (thus, you get a `provider` instead
    of a `container`).

    """

    name = "harp.config.bound"

    def __init__(self, provider: Services, resolver: ProxyControllerResolver):
        self.provider = provider
        self.resolver = resolver


EVENT_BOUND = OnBoundEvent.name

OnBoundHandler = Callable[[OnBoundEvent], Awaitable[None]]


class OnReadyEvent(Event):
    """
    «Ready» event happens after the asgi kernel is created, just before the application starts.

    It is the right place to decorate the kernel with middlewares.

    """

    name = "harp.config.ready"

    def __init__(self, provider: Services, kernel: ASGIKernel, binds: list[Bind]):
        self.provider = provider
        self.kernel = kernel
        self.binds = binds


EVENT_READY = OnReadyEvent.name

OnReadyHandler = Callable[[OnReadyEvent], Awaitable[None]]


class OnShutdownEvent(Event):
    """
    «Shutdown» event happens when the application is stopping.

    """

    name = "harp.config.shutdown"

    def __init__(self, kernel: ASGIKernel, provider: Services):
        self.kernel = kernel
        self.provider = provider


EVENT_SHUTDOWN = OnShutdownEvent.name

OnShutdownHandler = Callable[[OnShutdownEvent], Awaitable[None]]
