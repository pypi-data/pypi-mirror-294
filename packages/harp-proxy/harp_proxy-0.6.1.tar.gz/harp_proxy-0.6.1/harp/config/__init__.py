"""
The Config (:mod:`harp.config`) module provides tools to configure *Applications*.

*Applications* are standard python packages that contains specific definitions to be able to register services and
configure themselves. Every feature of Harp is built as an application, and you will want to write applications if you
want to extend harp's functionality for specific needs.


Contents
--------

"""

__title__ = "Config"

from .applications import Application, ApplicationsRegistry
from .builders import ConfigurationBuilder, System, SystemBuilder
from .events import (
    EVENT_BIND,
    EVENT_BOUND,
    EVENT_READY,
    EVENT_SHUTDOWN,
    OnBindEvent,
    OnBoundEvent,
    OnReadyEvent,
    OnShutdownEvent,
)
from .settings import (
    BaseSetting,
    Definition,
    DisableableBaseSettings,
    DisabledSettings,
    FromFileSetting,
    Lazy,
    asdict,
    settings_dataclass,
)
from .utils import get_application

__all__ = [
    "Application",
    "ApplicationsRegistry",
    "BaseSetting",
    "ConfigurationBuilder",
    "Definition",
    "DisableableBaseSettings",
    "DisabledSettings",
    "EVENT_BIND",
    "EVENT_BOUND",
    "EVENT_READY",
    "EVENT_SHUTDOWN",
    "FromFileSetting",
    "Lazy",
    "OnBindEvent",
    "OnBoundEvent",
    "OnReadyEvent",
    "OnShutdownEvent",
    "System",
    "SystemBuilder",
    "asdict",
    "get_application",
    "settings_dataclass",
]
