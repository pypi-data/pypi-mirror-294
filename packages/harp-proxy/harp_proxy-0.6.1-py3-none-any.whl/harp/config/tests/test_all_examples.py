import pytest

import harp
from harp.config.examples import (
    _get_available_documentation_examples_filenames,
    get_available_examples,
    get_example_filename,
)


def test_examples_list(snapshot):
    assert get_available_examples() == snapshot


def test_documentation_examples_list(snapshot):
    assert [x.removeprefix(harp.ROOT_DIR + "/") for x in _get_available_documentation_examples_filenames()] == snapshot


@pytest.mark.parametrize("example", get_available_examples())
def test_load_example(example):
    from harp.config import ConfigurationBuilder

    builder = ConfigurationBuilder()
    builder.add_file(get_example_filename(example))
    settings = builder.build()

    assert settings


@pytest.mark.parametrize("configfile", _get_available_documentation_examples_filenames())
def test_load_documentation_example(configfile):
    from harp.config import ConfigurationBuilder

    builder = ConfigurationBuilder()
    if "/apps/rules/" in configfile:
        builder.applications.add("rules")
    builder.add_file(configfile)
    settings = builder.build()
    applications = settings.pop("applications", [])
    assert len(set(settings.keys()).difference(set(builder.applications.keys()))) == 0
    assert len(applications) >= len(settings)
