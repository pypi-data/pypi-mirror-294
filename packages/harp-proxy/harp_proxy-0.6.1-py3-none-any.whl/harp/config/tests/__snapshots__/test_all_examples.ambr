# serializer version: 1
# name: test_documentation_examples_list
  list([
    'docs/apps/dashboard/examples/auth.basic.yml',
    'docs/apps/dashboard/examples/auth.yml',
    'docs/apps/dashboard/examples/devserver.yml',
    'docs/apps/dashboard/examples/main.yml',
    'docs/apps/http_client/examples/full.yml',
    'docs/apps/http_client/examples/simple.yml',
    'docs/apps/proxy/examples/swapi.yml',
    'docs/apps/rules/examples/rules.yml',
    'docs/apps/storage/examples/mysql-aiomysql.yml',
    'docs/apps/storage/examples/mysql-asyncmy.yml',
    'docs/apps/storage/examples/postgres-asyncpg.yml',
    'docs/apps/storage/examples/redis.yml',
    'docs/apps/storage/examples/sqlite-aiosqlite.yml',
  ])
# ---
# name: test_examples_list
  list([
    'auth.basic',
    'httpbin',
    'httpbins',
    'postgres',
    'repositories',
    'sentry',
    'sqlite',
  ])
# ---
